[doc] Add hpu resource description in ray train related docs (#47241)
ray-project/ray: doc/source/train/getting-started-pytorch.rst

Learn how to:

> 1. Configure a model to run distributed and on the correct CPU/GPU/other accelerator devices.

Reviewer: ```suggestion
1. Configure a model to run distributed and on the correct CPU, GPU, or other accelerator device.
```

> 2. Configure a dataloader to shard data across the :ref:`workers <train-overview-worker>` and place data on the correct CPU, GPU or other accelerator devices.

Reviewer 2: ```suggestion
2. Configure a dataloader to shard data across the :ref:`workers <train-overview-worker>` and place data on the correct CPU, GPU, or other accelerator devices.
```

3. Configure a :ref:`training function <train-overview-training-function>` to report metrics and save checkpoints.

> 4. Configure :ref:`scaling <train-overview-scaling-config>` and CPU, GPU, or :ref:`other accelerator <using-other-accelerators>` resource requirements for a training job.

Reviewer: Same comment as above.

```suggestion
4. Configure :ref:`scaling <train-overview-scaling-config>` and CPU, GPU, or other accelerator resource requirements for a training job.
```

    result = trainer.fit()

1. `train_func` is the Python code that executes on each distributed training worker.

> 2. :class:`~ray.train.ScalingConfig` defines the number of distributed training workers, and whether to use GPUs, CPUs or other types of accelerator devices.

Reviewer 2: ```suggestion
2. :class:`~ray.train.ScalingConfig` defines the number of distributed training workers, and whether to use CPUs, GPUs, or other types of accelerator devices.
```
